packages/expo-router/src/global-state/routing.ts

```diff
@@ -154,7 +154,7 @@ function getNavigateAction(state: ResultState, parentState: NavigationState, typ
 
   if (type === 'PUSH' && parentState.type !== 'stack') {
     type = 'NAVIGATE';
-  } else if (type === 'REPLACE' && parentState.type === 'tab') {
+  } else if (type === 'REPLACE' && parentState.type !== 'stack') {
```

Comment:
This will prevent custom navigators who might use the "REPLACE" type from working. We should only switch the type for Tabs and Drawers

Reply:
Good point, changing it!